### ISSUE TITLE
consumer key/secret question

### ISSUE THREAD

**Issue by rpedela** [domenic/restify-oauth2#1]
Great module! This is a question rather than an issue.

I am building an API where we want to give the API user a key and secret. Then they use that key and secret to authenticate with our server and then can use the API however they like. They are not required to ask a real user for their username and password.

I would like to know if they supply the key for the clientId and username and the secret for the clientSecret and password like below, is that acceptable for my purpose?

``` js
validateClient(API_KEY, API_SECRET, cb)
grantToken(API_KEY, API_SECRET, cb) 
```

In other words, can I abuse the password credentials flow or should I be using a completely different flow for this use case?
